src/main/python/cfn_sphere/exceptions.py

```diff
@@ -1,11 +1,24 @@
+from boto.exception import BotoServerError
+
+
 class CfnSphereException(Exception):
-    pass
+    def __init__(self, message="", boto_exception=None):
+        self.message = message
+        self.str = self.message
+        try:
+            self.request_id = boto_exception.request_id
+            self.str += " (Request ID: {0})".format(self.request_id)
+            self.boto_exception = boto_exception
+        except:
```

Comment:
[Asking for forgiveness is better than asking for permission](https://docs.python.org/2/glossary.html#term-eafp) the empty except is on purpose. If we cannot access the extra information then we simply fall back to the generic message.